posht/Posht.psm1

```diff
@@ -434,10 +488,68 @@ function RequestUriArgCompleter {
   $Requests | Where-Object { $_.GetUri() -like "$wordToComplete*" } | ForEach-Object { $_.GetUri() }
 }
 
+function HashtableToString {
+  param ( 
+    [hashtable]$Hashtable
+  )
+
+  if ($null -eq $Hashtable) {
+    return "";
+  }
+  
+  $Result = ""
+
+  $Hashtable.Keys | ForEach-Object { $Result += "$_=$($Hashtable[$_])," }
+  return $Result
+}
+
 #endregion
 
 #region public functions
 
+<#
+.SYNOPSIS
+Creates a new posht config/request file (posht.json)
+
+.DESCRIPTION
+Creates a new posht config/request file (posht.json)
+If the local switch is not specified, file is saved in the user profile path
+
+.PARAMETER Local
+Create and save at current path (instead of user profile path)
+
+.EXAMPLE
+New-ApiConfig
+New-ApiConfig -Local
+
+#>
+function New-ApiConfig {
+  [CmdletBinding(SupportsShouldProcess = $true, ConfirmImpact = 'High')]
+  param (
+    [Parameter(Mandatory = $false)]
+    [switch]$Local
+  )
+
+  $FullPath = ""
+  if($Local){
+    $FullPath = Get-ApiConfigFilePath -Source Local
+  } else {
+    $FullPath = Get-ApiConfigFilePath -Source UserProfile 
+  }
+
+  if ($PSCmdlet.ShouldProcess("This will create a new config file under '$FullPath'. If there is an existing one, it will be overwritten!", $FullPath, 'Initialize new config file (may overwrite exting one)?')) {
```

Comment:
I have a small improvement suggestion; when the config file does not exists the confirmation can be omitted. This can be achieved setting confirmpreference to none:

```powershell
  if (-not (Test-Path $FullPath)) {
    $ConfirmPreference = 'None' 
  }
```

Also there is a typo: exting -> existing.

Reply:
Is the ConfirmPreference then only set to 'None' for the current function or do I have to reset it somehow?

Reply:
Ok, is fixed. I just extended the if

Reply:
Oh yes thats better.